pkg/mock/mock.go

```diff
@@ -0,0 +1,72 @@
+package mock
+
+import (
+	"path"
+	"reflect"
+	"sync"
+
+	"github.com/pingcap/failpoint"
+)
+
+// Finalizer represent the function that clean a mock point
+type Finalizer func() error
+
+type mockPoints struct {
+	m map[string]interface{}
+	l sync.Mutex
+}
+
+func (p *mockPoints) set(fpname string, value interface{}) {
+	p.l.Lock()
+	defer p.l.Unlock()
+
+	p.m[fpname] = value
+}
+
+func (p *mockPoints) get(fpname string) interface{} {
+	p.l.Lock()
+	defer p.l.Unlock()
+
+	return p.m[fpname]
+}
+
+func (p *mockPoints) clr(fpname string) {
+	p.l.Lock()
+	defer p.l.Unlock()
+
+	delete(p.m, fpname)
+}
+
+var points = mockPoints{m: make(map[string]interface{})}
```

Comment:
I don't thinks we have to share variables with failpoint injection.

In my opinion, if you want to inject multiple different errors, using multiple failpoint seems a better idea.

Reply:
I agree with the keao

Reply:
I don't understand very well, how to archive this? Can you write a tiny demo? We need inject mock values among packages and the values can be any type...

Reply:
Well! I agree that there is no better way. It's ok for me.